Automattic/pocket-casts-android
# Extract media service out of sharing

## Description

Follow up to https://github.com/Automattic/pocket-casts-android/pull/2707#pullrequestreview-2259384822.

> The Wear OS APK is 136MB. I understand it will be smaller in the final version, but is it possible to exclude the feature from the Wear OS version as it isn't used, or is this too tricky?

This PR required restructuring sharing code. You can see it in a diagram below.

| Before | After |
| - | - |
| ![before](https://github.com/user-attachments/assets/deeb218b-3803-4e7c-a70a-1e28292222bf) | ![after](https://github.com/user-attachments/assets/21c9c5ef-e914-4dbd-80ef-87006e505b12) |

## Testing Instructions

1. Test clip sharing.
2. Build the Wear app. It should not include FFmpeg.
3. Build the Automotive app. It should not include FFmpeg.

## Checklist
 
#### I have tested any UI changes...
